internal/status/gatewaystatus.go

```diff
@@ -93,7 +108,15 @@ func getGatewayConditions(gs *gatewayapi_v1alpha2.GatewayStatus) map[gatewayapi_
 	return conditions
 }
 
-func (gatewayUpdate *GatewayConditionsUpdate) Mutate(obj client.Object) client.Object {
+func (gatewayUpdate *GatewayStatusUpdate) Mutate(obj client.Object) client.Object {
+	o, ok := obj.(*gatewayapi_v1alpha2.Gateway)
+	if !ok {
+		panic(fmt.Sprintf("Unsupported %T object %s/%s in GatewayStatusUpdate status mutator",
```

Comment:
I commented when this was first added, but I still dislike the `panic(..)` here. I don't feel that status failing to update should cause Contour to hard crash. :man_shrugging:

Reply:
I have no objection to switching the Mutate funcs to return an error instead of panicking. In reality this code branch really _shouldn't_ ever get hit in a released version of Contour, it would require there to be a bug in the status code that would cause other tests to fail too, but never say never.

Reply:
Or we can wait until Go 1.18 and use generics :trollface:

Reply:
Yeah, the reason I put the panic in there is because there's already a few layers of defense against unsupported objects getting into a mutator. It should never happen, and in the unlikely event that it does, we want it to break tests or something.